Updated to CMS 6 / 6.2

StevenPaw opened the pull request:
Updated Classnames, changed init to OnInit and cleaned up the code a bit for usage in Silverstripe CMS 6.0

StevenPaw committed:
Update composer.json to SS6.0

StevenPaw committed:
[~] Updated Classnames

StevenPaw committed:
[~] Updated onInit and some code quality stuff

thats4shaw commented:
This works great apart from the grouped menu functionality no longer working.

Some inspiration can probably be gleaned from https://github.com/bigfork/silverstripe-grouped-cms-menu-fork in regards to the restoration of the javascript removed from core.

rosis-sharma-trainor committed:
Restore grouped menu JS

rosis-sharma-trainor commented:
I’ve opened a pull request on the forked repository that addresses the grouped menu issue. @StevenPaw Please take a look: https://github.com/StevenPaw/silverstripe-cms-theme/pull/1.

StevenPaw committed:
Merge pull request #1 from rosis-sharma-trainor/master

Restore grouped menu JS

StevenPaw commented:
The pull request of @rosis-sharma-trainor was approved and now the grouped menus also work properly again.

StevenPaw committed:
[~] Reenabled Toolbar South

StevenPaw committed:
[~] Fixed styling of Toolbar South

ntd commented:
Thank your for your work!

Using Silverstripe 6.2.0 without CMS and your master branch (ac9a5edcc0c77b3dbf9e5be858a46d3adeab1eea), I'm experiencing padding issues. The two main problems are:
1. the login link is padded differently than the rest
2. `.cms-menu__list li.opened a` has a different padding-left (5px vs 26px)

In my YAML I'm just customizing colors, i.e.:
```yaml
SilverStripe\Admin\LeftAndMain:
  cms_background: '#008a00'
  cms_border_color: '#368b39'
  cms_color: '#fff'
  cms_hover_background: '#007a00'
  cms_hover_color: currentColor
  cms_active_background: '#006a00'
  cms_active_color: currentColor
  cms_drawer_background: '#1a883e'
  cms_drawer_color: currentColor
  cms_icon_color: currentColor
```
Here are the unbranded vs branded screenshots:
<img width="415" height="453" alt="paddings" src="https://github.com/user-attachments/assets/dac4cc34-bde4-4b65-9c09-85b55d4f9717" />

I'm just wondering if I'm the only one with those issues.

thats4shaw commented:
@ntd there were some changes that came through in 6.2.0 in regards to CMS menu.

The below will work around it:
```
// The SS admin CSS sets padding-left: 5px on all .opened a descendants
// (specificity 0,2,2), overriding the cms-theme's padding (0,1,1).
// This collapses text into icons. Restore proper padding for both
// the parent group link and the child items inside.
.cms-menu__list .branded-menu__list-item.opened > a {
  padding-left: 2rem !important;
}

.cms-menu__list .branded-menu__list-item.opened .branded-menu--child a {
  padding-left: 2.5rem !important;
}

// Remove the default left border on opened/current menu items.
.cms-menu__list .opened {
  border-left: none !important;
}
```

ntd commented:
> The below will work around it:
> ...

Yes, that fixes the second problem (the wrong left padding of the selected item).

StevenPaw committed:
[~] Changed to vite and added cms-info-option

StevenPaw commented:
I added some changes today because the pull request haven't been answered yet and there were some requests from other people for a switch to display or not display the cms info menu on the bottom left. While at it, i also fixed the wrong spacing and made the login info a bit smaller than the other menu items to make it visible different from modeladmins. This version now works with Silverstripe 6.2.
I hope this helps ^^
